### PR TITLE
Page perf metrics

### PR DESCRIPTION
**Pull request by carlesandres**
This adds the ability to track [built-in browser navigation time marks](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming) from within `o-ads`. 

It's going to be useful to compare ads-loading performance with overall page-loading performance.

### CODE REVIEW

File: src/js/utils/metrics.js [hunk 1]
```diff
@@ -20,6 +21,25 @@ function getMarksForEvents(events, suffix) {
 	return marks;
 }
 
+function getNavigationPerfMarks(desiredMarks) {
+	if (!performance || !performance.getEntriesByType || !utils.isArray(desiredMarks)) {
```
Comment (alevito) [inline src/js/utils/metrics.js]: I can see `performance` initialised within `getMarksForEvents()` but not within this function. Is it within a global variable?

Comment (alevito) [reply src/js/utils/metrics.js]: oops, forget about this. I just noticed it is a draft PR

Comment (carlesandres) [reply src/js/utils/metrics.js]: That's actually a good finding. `performance` is a global object defined in the DOM. I've done some research and it seems to be supported by all the main browsers we care about so I will not re-define it.

File: src/js/utils/metrics.js [hunk 2]
```diff
@@ -2,7 +2,6 @@ import utils from './index';
 
 function getMarksForEvents(events, suffix) {
 	const markNames = events.map( eventName => 'oAds.' + eventName + suffix );
-	const performance = window.LUX || window.performance || window.msPerformance || window.webkitPerformance || window.mozPerformance;
```
Comment (alexflorisca) [inline src/js/utils/metrics.js]: What is window.LUX and should we be removing it?

Comment (carlesandres) [reply src/js/utils/metrics.js]: LUX is [SpeedCurve's library](https://support.speedcurve.com/articles/867066-lux-api). I took the idea of using it from [this bit of code in n-ui-foundations](https://github.com/Financial-Times/n-ui-foundations/blob/a050039ab0cc29241f7258cf567504ff9781d05e/js/perf-mark.js) but I can see it's not actually being used on ft.com so I think we can remove it.

I have also found that the native 'performance' global object is [well supported](https://developer.mozilla.org/en-US/docs/Web/API/Performance#Browser_compatibility) across most of the browsers we care about.

**Pull request merged**